hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/zookeeper/ZookeeperRepositoryManager.java

```diff
@@ -0,0 +1,128 @@
+        for (ZookeeperClient client : clientManager.getClients()) {
+            String dcName = client.getDcName();
+            CuratorFramework zookeeper = client.getCuratorFramework();
+
+            GroupRepository groupRepository = new ZookeeperGroupRepository(zookeeper, mapper, paths);
+            TopicRepository topicRepository = new ZookeeperTopicRepository(zookeeper, mapper, paths, groupRepository);
+            SubscriptionRepository subscriptionRepository = new ZookeeperSubscriptionRepository(zookeeper, mapper,
+                    paths, topicRepository);
+            OAuthProviderRepository oAuthProviderRepository = new ZookeeperOAuthProviderRepository(zookeeper, mapper,
+                    paths);
+            SubscriptionOffsetChangeIndicator offsetChangeIndicator =
+                    new ZookeeperSubscriptionOffsetChangeIndicator(zookeeper, paths, subscriptionRepository);
+            MessagePreviewRepository messagePreviewRepository = new ZookeeperMessagePreviewRepository(zookeeper, mapper,
+                    paths);
+            TopicBlacklistRepository topicBlacklistRepository = new ZookeeperTopicBlacklistRepository(zookeeper, mapper,
+                    paths);
+            UndeliveredMessageLog undeliveredMessageLog = new ZookeeperUndeliveredMessageLog(zookeeper, paths, mapper);
+
+            groupRepositoriesByDc.put(dcName, groupRepository);
+            topicRepositoriesByDc.put(dcName, topicRepository);
+            subscriptionRepositoriesByDc.put(dcName, subscriptionRepository);
+            oAuthProviderRepositoriesByDc.put(dcName, oAuthProviderRepository);
+            offsetChangeIndicatorsByDc.put(dcName, offsetChangeIndicator);
+            messagePreviewRepositoriesByDc.put(dcName, messagePreviewRepository);
+            topicBlacklistRepositoriesByDc.put(dcName, topicBlacklistRepository);
+            undeliveredMessageLogsByDc.put(dcName, undeliveredMessageLog);
+        }
+    }
+
+    public <T> DcBoundRepositoryHolder<T> getLocalRepository(Class<T> repositoryType) {
+        String dcName = dcNameProvider.getDcName();
+        T repository = getRepositoriesByType(repositoryType).get(dcName);
+
+        if (repository == null) {
+            throw new InternalProcessingException("Failed to find '" + repositoryType.getSimpleName() +
+                    "' bound with DC '" + dcName + "'.");
+        }
+
+        return new DcBoundRepositoryHolder<>(repository, dcName);
+    }
+
+    public <T> List<DcBoundRepositoryHolder<T>> getRepositories(Class<T> repositoryType) {
+        return getRepositoriesByType(repositoryType)
+                .entrySet()
+                .stream()
+                .map(entry -> new DcBoundRepositoryHolder<>(entry.getValue(), entry.getKey()))
+                .collect(Collectors.toList());
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> Map<String, T> getRepositoriesByType(Class<T> type) {
+        if (type == GroupRepository.class) {
```

Comment:
We can also put these repositories in Map<Class<?>, Object> and get them by class key. During the get operation we will have to cast value to (Map<String, T>) but thanks to this we will be able to remove this big block of ifs.